### PR TITLE
Change the classifier input from 2048 to 1000.

### LINKED ISSUE #5231: Typo correction in Transfer Learning sample code
I tried to write the code by referring to the [Transfer Learning page](https://pytorch-lightning.readthedocs.io/en/stable/transfer_learning.html#example-imagenet-computer-vision) on readthedocs, but I got `RuntimeError: mat1 dim 1 must match mat2 dim 0`.

Since the FC layer of RenNet50 is a classification of 1000 classes, I tried setting the input of `self.classifier` to 1000, and it worked.

I am new to machine learning, so I am not sure if this is really the right fix. Thanks!

### PR DESCRIPTION
**Pull request by LaserBit**
## What does this PR do?

Fixes #5231

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by rohitgr7** [comment]
```python
class ImagenetTransferLearning(LightningModule):
    def __init__(self):
        super().__init__()

        # init a pretrained resnet
        backbone = models.resnet50(pretrained=True)
        num_filters = backbone.fc.in_features
        _layers = list(backbone.children())[:-1]
        self.feature_extractor = torch.nn.Sequential(*_layers)

        # use the pretrained model to classify cifar-10 (10 image classes)
        num_target_classes = 10
        self.classifier = nn.Linear(num_filters, num_target_classes)

    def forward(self, x):
        self.feature_extractor.eval()
        with torch.no_grad():
        	representations = self.feature_extractor(x)
        x = self.classifier(representations)
        ...
```

I think it should be like this.
1. Last layer of pretrained model is basically a classifier and putting  new classifier on top of it is a bit incorrect.
2. `LightningModule` puts the entire model to train mode while training. So need to manually do `.eval` everytime in forward + `no_grad` to avoid gradient update.

**Comment by LaserBit** [comment]
Thank you !
However, just like the original sample code, I could not use your code because I got `RuntimeError: mat1 dim 1 must match mat2 dim 0`.
I tried using the CIFAR-10 dataset to see if my own dataset was the problem, but I got the same `RuntimeError: mat1 dim 1 must match mat2 dim 0`.

This is my source code. [Google Colab](https://colab.research.google.com/drive/1Sq_L8fB1W9y0I4w4SAjCnwh-Julih0Wz)

Can you give me a solution?

**Comment by rohitgr7** [comment]
@LaserBit fixed it. should be working now.

**Comment by LaserBit** [comment]
@rohitgr7 Thanks!
It's working fine, sir.

**Review by rohitgr7 — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Review by Borda — APPROVED** [review]
lgtm

File: docs/source/transfer_learning.rst [hunk 1]
```diff
@@ -52,16 +52,23 @@ Example: Imagenet (computer Vision)
...
-            self.classifier = nn.Linear(2048, num_target_classes)
+            num_target_classes = 10
+            self.classifier = nn.Linear(num_filters, num_target_classes)
 
         def forward(self, x):
-            representations = self.feature_extractor(x)
+            self.feature_extractor.eval()
+            batch_size = x.size(0)
+            with torch.no_grad():
+                representations = self.feature_extractor(x).view(batch_size, -1)
```
Comment (awaelchli) [inline docs/source/transfer_learning.rst:71]:
```suggestion
            with torch.no_grad():
                representations = self.feature_extractor(x).flatten(1)
```
let's use flatten here?

File: docs/source/transfer_learning.rst [hunk 2]
```diff
@@ -52,16 +52,23 @@ Example: Imagenet (computer Vision)
...
+            super().__init__()
+
             # init a pretrained resnet
-            num_target_classes = 10
-            self.feature_extractor = models.resnet50(pretrained=True)
-            self.feature_extractor.eval()
+            backbone = models.resnet50(pretrained=True)
+            num_filters = backbone.fc.in_features
+            _layers = list(backbone.children())[:-1]
+            self.feature_extractor = torch.nn.Sequential(*_layers)
```
Comment (awaelchli) [inline docs/source/transfer_learning.rst:61]:
```suggestion
            layers = list(backbone.children())[:-1]
            self.feature_extractor = torch.nn.Sequential(*layers)
```
it seems arbitrary that some vars here are prefixed with underscore

**Review by awaelchli — APPROVED** [review]

File: docs/source/transfer_learning.rst (hunk 1 above)
Comment (awaelchli) [reply docs/source/transfer_learning.rst:71]: perhaps it's not even necessary, as the goal of these docs is probably just to show the pseudo code and get the idea across :)

**Review by SkafteNicki — APPROVED** [review]
LGTM

**Pull request merged**